test/framework/kogitoserviceutils.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/meta"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/framework"
+	"github.com/kiegroup/kogito-cloud-operator/test/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// WaitForService waits that the service has a certain number of replicas
+func WaitForService(namespace string, serviceName string, replicas int, timeoutInMin int) error {
+	return WaitForOnOpenshift(namespace, serviceName+" running", timeoutInMin,
+		func() (bool, error) {
+			deployment, err := GetDeployment(namespace, serviceName)
+			if err != nil {
+				return false, err
+			}
+			if deployment == nil {
+				return false, nil
+			}
+			return deployment.Status.Replicas == int32(replicas) && deployment.Status.AvailableReplicas == int32(replicas), nil
+		})
+}
+
+// CustomResourceBuilder helps to configure a CR resource used by the InstallService method.
+type CustomResourceBuilder struct {
+	GetResource       meta.ResourceObject
+	SetObject         func(object metav1.ObjectMeta)
+	SetSpec           func(spec v1alpha1.KogitoServiceSpec)
+	SetStatus         func(status v1alpha1.KogitoServiceStatus)
+	EnablePersistence func()
+}
+
+// InstallService the Kogito Service component
+func InstallService(namespace string, serviceName string, replicas int, installerType InstallerType, imageTag string, cliName string, crBuilder CustomResourceBuilder, persistence bool) error {
+	GetLogger(namespace).Infof("%s install %s with %d replicas and persistence %v", serviceName, installerType, replicas, persistence)
+	image := buildImage(imageTag)
+	switch installerType {
+	case CLIInstallerType:
+		return cliInstall(namespace, replicas, image, cliName, persistence)
+	case CRInstallerType:
+		return crInstall(namespace, serviceName, int32(replicas), image, crBuilder, persistence)
+	default:
+		panic(fmt.Errorf("Unknown installer type %s", installerType))
+	}
+}
+
+func buildImage(imageTag string) v1alpha1.Image {
+	image := framework.ConvertImageTagToImage(imageTag)
+	if len(image.Tag) == 0 || image.Tag == "latest" {
```

Comment:
this won't work. you set the `services-image-version` if it is not empty and if there is no `data-index-image-tag` or `jobs-service-image-tag` or `management-console-image-tag` defined (depending on the service you want or install)

The default image full tag has anyway a Tag, so it would not be overriden here
If you defined a specific service image tag, then it could be `latest` tag in it and you would still override it whereas you should not.

Reply:
we could have a `imageFullTag` (empty if no service image tag defined) and a `defautImageFullTag`

Reply:
I reverted this change.